crates/walrus-sdk/src/client/quilt_client.rs

```diff
@@ -0,0 +1,583 @@
+        blobs_with_paths: &[(PathBuf, Vec<u8>)],
+    ) -> Vec<QuiltStoreBlob> {
+        blobs_with_paths
+            .iter()
+            .enumerate()
+            .map(|(i, (path, blob))| {
+                QuiltStoreBlob::new(
+                    blob,
+                    path.file_name()
+                        .and_then(|file_name| file_name.to_str())
+                        .map(String::from)
+                        .unwrap_or_else(|| format!("unnamed-blob-{}", i)),
+                )
+            })
+            .collect()
+    }
+
+    /// Constructs a quilt from a list of paths.
+    ///
+    /// The paths can be files or directories; if they are directories, their files are read
+    /// recursively.
+    //
+    /// The on-disk file names are used as identifiers for the quilt patches.
+    /// If the file name is not valid UTF-8, it will be replaced with "unnamed-blob-index".
+    pub async fn construct_quilt_from_paths<V: QuiltVersion, P: AsRef<Path>>(
+        &self,
+        paths: &[P],
+        encoding_type: EncodingType,
+    ) -> ClientResult<V::Quilt> {
+        let blobs_with_paths = read_blobs_from_paths(paths)?;
+        if blobs_with_paths.is_empty() {
+            return Err(ClientError::from(ClientErrorKind::Other(
+                "No valid files found in the specified folder".into(),
+            )));
+        }
+
+        let quilt_store_blobs: Vec<_> = Self::assign_identifiers_with_paths(&blobs_with_paths);
+
+        self.construct_quilt::<V>(&quilt_store_blobs, encoding_type)
+            .await
+    }
+
+    /// Stores all blobs from a list of paths as a quilt.
+    #[tracing::instrument(skip_all)]
+    pub async fn reserve_and_store_quilt_from_paths<V: QuiltVersion, P: AsRef<Path>>(
+        &self,
+        paths: &[P],
+        encoding_type: EncodingType,
+        epochs_ahead: EpochCount,
+        store_when: StoreWhen,
+        persistence: BlobPersistence,
+        post_store: PostStoreAction,
+    ) -> ClientResult<QuiltStoreResult> {
+        // Read blobs from the paths.
+        let blobs_with_paths = read_blobs_from_paths(paths)?;
+        if blobs_with_paths.is_empty() {
+            return Err(ClientError::from(ClientErrorKind::Other(
+                "No valid files found in the specified folder".into(),
+            )));
+        }
+
+        let quilt_store_blobs: Vec<_> = Self::assign_identifiers_with_paths(&blobs_with_paths);
+
+        let result = self
+            .reserve_and_store_quilt::<V>(
+                &quilt_store_blobs,
+                encoding_type,
+                epochs_ahead,
+                store_when,
+                persistence,
+                post_store,
+            )
+            .await?;
+
+        Ok(result)
+    }
+
+    /// Encodes the blobs to a quilt and stores it to Walrus.
+    #[tracing::instrument(skip_all, fields(blob_id))]
+    pub async fn reserve_and_store_quilt<V: QuiltVersion>(
+        &self,
+        quilt_store_blobs: &[QuiltStoreBlob<'_>],
```

Comment:
Suggestion: Change this to `V::Quilt` (removing the quilt construction below), then change the `reserve_and_store_quilt_from_paths` to just call `construct_quilt_from_paths` and `reserve_and_store_quilt`.